Add best practices for STAC Zarr (#29)
radiantearth/stac-best-practices: best-practices-zarr.md

- **Group**: Any node - including the root node - in a Zarr hierarchy that may have child nodes.
- **Array**: A leaf node in a Zarr hierarchy that represents a multidimensional array with chunked storage
- **Chunk**: A contiguous region of an array that is stored as a single object

Conventions terms:

- **Multiscales Group**: A [convention](https://github.com/zarr-conventions/multiscales) for organizing multi-resolution data in Zarr stores, typically using a tree structure where each level represents a different resolution.

### Xarray/NetCDF/CF-Specific Concepts

- **Variable**: A named data array within a Zarr group, typically representing a measured or derived quantity
- **Dimension**: An axis of a multidimensional array (e.g., x, y, time, band)
- **Coordinate Variable**: An array that provides coordinate values along a dimension
- **Data Variable**: An array containing measured or derived data values

## General Rules

### STAC Object Type

> Zarr stores and n-dimensional arrays can be represented by 2 STAC object types depending on the use case.

Reviewer: ```suggestion
Zarr groups and arrays can be represented by 2 STAC object types depending on the use case.
```

#### STAC Item

A STAC Item is appropriate when the Zarr store or group represents a single logical dataset or scene, such as:

- A satellite scene with multiple bands and resolutions
- A time-specific climate model run output with multiple variables and dimensions
- A single time slice of a time series
- A single multiscale dataset

> It is **HIGHLY RECOMMENDED** to reference a unique Zarr store per STAC Item to avoid ambiguity when accessing data via the asset hrefs.

Reviewer: This meaning of this line isn't clear to me.

Author: The reference is the recommended `link` to the zarr store. The idea is to discourage having Stac Item with assets pointing to 2 different Zarr store.

dataset.zarr/
      │
      ├─► measurements/
      │   │
      │   └─► reflectance/         (multiscales group)
      │        │
      │        ├─► r10m/           (resolution level)
      │        │    ├─► b02/       (array)
      │        │    ├─► b03/       (array)
      │        │    └─► b04/       (array)
      │        │
      │        ├─► r20m/           (resolution level)
      │        │    └─► ...
      │        │
      │        ├─► r60m/           (resolution level)
      │        │    └─► ...
      │
      └─► quality/
```

The `bands` array SHALL reference individual bands, and the `name` field SHOULD follow the same pattern as in the "Unique band per data variable" and "Multiple bands per data variable" cases, depending on how bands are organized within each resolution group.
The key point is that the resolution group do not appear directly in the metadata but is implicitly hidden behind the asset href. It is up to the client to construct and select the correct path to access individual arrays based on the [layout](https://github.com/zarr-conventions/multiscales?tab=readme-ov-file#layout) given in the multiscales convention.

```json
"assets": {
    "reflectance": {

>       "href": "s3://esa-zarr-sentinel-explorer-fra/tests-output/sentinel-2-l2a/S2A_MSIL2A_20251008T100041_N0511_R122_T32TQM_20251008T122613.zarr/measurements/reflectance",

Reviewer: Similar to the suggestion above, I think the zarr store location should be consistent across these examples:
```suggestion
      "href": "s3://bucket/path/data.zarr/measurements/reflectance",
```

```json
"linkTemplates": [
  {
    "rel": "data-variable",
    "title": "r10m",
    "uriTemplate": "s3://bucket/path/data.zarr/group/{band}",
    "variables": {
      "band": {
        "description": "...",
        "type": "string",
        "enum": [
          "B02",
          "B03",
          "B04",
        ]
      }
    }
  }
]
```

## STAC Extensions Requirements

1. **[Datacube extension](https://github.com/stac-extensions/datacube) >=v2.0.0 SHOULD be used to describe variables and dimensions when relevant**

   It is used to describe multidimensional structure for groups containing multiple variables, use `cube:variables` and `cube:dimensions`, especially when variable arrays have more than the 2 spatial dimensions.

   - Available variables and their dimensional structure
   - Dimension extents and coordinate systems
   - Relationships between variables

2. **[Projection Extension](https://github.com/stac-extensions/projection) >=v2.0.0 SHOULD be used to describe spatial reference information**

   Key attributes to include:

   - `proj:code`, `proj:wkt2`, or `proj:projjson`
   - `proj:shape` and `proj:transform`

>   If the referenced Zarr store follows the [geo-proj convention](https://github.com/zarr-conventions/geo-proj), the projection information should be consistent with the convention.

Reviewer: ```suggestion
  If the referenced Zarr group follows the [geo-proj convention](https://github.com/zarr-conventions/geo-proj), the projection information should be consistent with the convention.
```

Virtual Zarr stores can be represented in both STAC Items and Collections, following the same patterns described in the [STAC Object Type](#stac-object-type) section:

- **STAC Item**: Appropriate when the virtual Zarr store represents a single logical dataset (e.g., a single time slice or scene)
- **STAC Collection**: Appropriate when organizing multiple related virtual Zarr stores (e.g., time series) or when a single virtual Zarr store contains a time dimension spanning the entire collection

The [example STAC Item](examples/CMIP6_ScenarioMIP_NCAR_CESM2.json) represents a virtual Zarr store for a single dataset.

**Key Points**:

- The link with `rel: store` SHOULD point to the entrypoint of the virtual Zarr store (e.g., a reference file)
- Assets with role `"data"` reference the Zarr groups/arrays accessible through the virtual store

> - Optionally, implementers MAY include assets with role `"source"` to reference the underlying data files

Reviewer: ```suggestion
- Optionally, implementers MAY include assets with role `"source"` to reference the bucket/directory containing the underlying data files
```
